File: spec/subtend/constants_spec.rb
require File.dirname(__FILE__) + '/../spec_helper'
require File.dirname(__FILE__) + '/subtend_helper'

compile_extension('subtend_constants')
require File.dirname(__FILE__) + '/ext/subtend_constants'

describe "SubtendConstants" do
  before :each do
    @s = SubtendConstants.new
  end

  it "rb_cObject should return the Object class" do
    @s.rb_object.should == Object
  end

  it "rb_mKernel should return the Kernel class" do
    @s.rb_kernel.should == Kernel
  end

  it "rb_eException should return the Exception class" do
    @s.rb_exception.should == Exception
  end

  it "rb_eSystemExit should return the SystemExit class" do
    @s.rb_system_exit.should == SystemExit
  end

  it "rb_eInterrupt should return the Interrupt class" do
    @s.rb_interrupt.should == Interrupt
  end

  it "rb_eSignal should return the SignalException class" do
    @s.rb_signal.should == SignalException
  end

  it "rb_eStandardError should return the StandardError class" do
    @s.rb_standard_error.should == StandardError
  end

  it "rb_eRuntimeError should return the RuntimeError class" do
    @s.rb_runtime_error.should == RuntimeError
  end

  it "rb_eTypeError should return the TypeError class" do
    @s.rb_type_error.should == TypeError
  end

  it "rb_eArgError should return the ArgumentError class" do
    @s.rb_arg_error.should == ArgumentError
  end

  it "rb_eIndexError should return the IndexError class" do
    @s.rb_index_error.should == IndexError
  end

  it "rb_eRangeError should return the RangeError class" do
    @s.rb_range_error.should == RangeError
  end

  it "rb_eNameError should return the NameError class" do
    @s.rb_name_error.should == NameError
  end

  it "rb_eNoMethodError should return the NoMethodError class" do
    @s.rb_no_method_error.should == NoMethodError
  end

  it "rb_eSecurityError should return the SecurityError class" do
    @s.rb_security_error.should == SecurityError
  end

  it "rb_eNotImpError should return the NotImplementedError class" do
    @s.rb_not_imp_error.should == NotImplementedError
  end

  it "rb_eNoMemError should return the NoMemoryError class" do
    @s.rb_no_mem_error.should == NoMemoryError
  end

  it "rb_eScriptError should return the ScriptError class" do
    @s.rb_script_error.should == ScriptError
  end

  it "rb_eSyntaxError should return the SyntaxError class" do
    @s.rb_syntax_error.should == SyntaxError
  end

  it "rb_eLoadError should return the LoadError class" do
    @s.rb_load_error.should == LoadError
  end

  it "rb_eSystemCallError should return the SystemCallError class" do
    @s.rb_system_call_error.should == SystemCallError
  end

  it "rb_mErrno should return the Errno class" do
    @s.rb_errno.should == Errno
  end

  it "rb_eIOError should return the IOError class" do
    @s.rb_io_error.should == IOError
  end

  it "rb_eLocalJumpError should return the LocalJumpError class" do
    @s.rb_local_jump_error.should == LocalJumpError
  end

  it "rb_eEOFError should return the EOFError class" do
    @s.rb_eof_error.should == EOFError
  end

  it "rb_eThreadError should return the ThreadError class" do
    @s.rb_thread_error.should == ThreadError
  end

  it "rb_eZeroDivError should return the ZeroDivisionError class" do
    @s.rb_zero_division_error.should == ZeroDivisionError
  end

  it "rb_eFloatDomainError should return the FloatDomainError class" do
    @s.rb_float_domain_error.should == FloatDomainError
  end

  it "rb_eSysStackError should return the SystemStackError class" do
    @s.rb_sys_stack_error.should == SystemStackError
  end

  it "rb_eRegexpError should return the RegexpError class" do
    @s.rb_regexp_error.should == RegexpError
  end
end

File: spec/capi/ext/constants_spec.c
#include <ruby.h>

VALUE sc_get_object (VALUE self) { 
  return rb_cObject;
}

VALUE sc_get_kernel (VALUE self) { 
  return rb_mKernel; 
}

VALUE sc_get_exception (VALUE self) {
  return rb_eException;
}

VALUE sc_get_system_exit (VALUE self) {
  return rb_eSystemExit;
}

VALUE sc_get_interrupt (VALUE self) {
  return rb_eInterrupt;
}

VALUE sc_get_signal (VALUE self) {
  return rb_eSignal;
}

VALUE sc_get_standard_error (VALUE self) {
  return rb_eStandardError;
}

VALUE sc_get_runtime_error (VALUE self) {
  return rb_eRuntimeError;
}

VALUE sc_get_type_error (VALUE self) {
  return rb_eTypeError;
}

VALUE sc_get_arg_error (VALUE self) {
  return rb_eArgError;
}

VALUE sc_get_index_error (VALUE self) {
  return rb_eIndexError;
}

VALUE sc_get_range_error (VALUE self) {
  return rb_eRangeError;
}

VALUE sc_get_name_error (VALUE self) {
  return rb_eNameError;
}

VALUE sc_get_no_method_error (VALUE self) {
  return rb_eNoMethodError;
}

VALUE sc_get_security_error (VALUE self) {
  return rb_eSecurityError;
}

VALUE sc_get_not_imp_error (VALUE self) {
  return rb_eNotImpError;
}

VALUE sc_get_no_mem_error (VALUE self) {
  return rb_eNoMemError;
}

VALUE sc_get_script_error (VALUE self) {
  return rb_eScriptError;
}

VALUE sc_get_syntax_error (VALUE self) {
  return rb_eSyntaxError;
}

VALUE sc_get_load_error (VALUE self) {
  return rb_eLoadError;
}

VALUE sc_get_system_call_error (VALUE self) {
  return rb_eSystemCallError;
}

VALUE sc_get_errno (VALUE self) {
  return rb_mErrno;
}

VALUE sc_get_io_error (VALUE self) {
  return rb_eIOError;
}

VALUE sc_get_local_jump_error (VALUE self) {
  return rb_eLocalJumpError;
}

VALUE sc_get_eof_error (VALUE self) {
  return rb_eEOFError;
}

VALUE sc_get_thread_error (VALUE self) {
  return rb_eThreadError;
}

VALUE sc_get_zero_division_error (VALUE self) {
  return rb_eZeroDivError;
}

VALUE sc_get_float_domain_error (VALUE self) {
  return rb_eFloatDomainError;
}

VALUE sc_get_sys_stack_error (VALUE self) {
  return rb_eSysStackError;
}

VALUE sc_get_regexp_error (VALUE self) {
  return rb_eRegexpError;
}

void Init_constants_spec() {
  VALUE cls;
  cls = rb_define_class("CApiConstantsSpecs", rb_cObject);
  rb_define_method(cls, "rb_object", sc_get_object, 0);
  rb_define_method(cls, "rb_kernel", sc_get_kernel, 0);
  rb_define_method(cls, "rb_exception", sc_get_exception, 0);
  rb_define_method(cls, "rb_system_exit", sc_get_system_exit, 0);
  rb_define_method(cls, "rb_interrupt", sc_get_interrupt, 0);
  rb_define_method(cls, "rb_signal", sc_get_signal, 0);
  rb_define_method(cls, "rb_standard_error", sc_get_standard_error, 0);
  rb_define_method(cls, "rb_runtime_error", sc_get_runtime_error, 0);
  rb_define_method(cls, "rb_type_error", sc_get_type_error, 0);
  rb_define_method(cls, "rb_arg_error", sc_get_arg_error, 0);
  rb_define_method(cls, "rb_index_error", sc_get_index_error, 0);
  rb_define_method(cls, "rb_range_error", sc_get_range_error, 0);
  rb_define_method(cls, "rb_name_error", sc_get_name_error, 0);
  rb_define_method(cls, "rb_no_method_error", sc_get_no_method_error, 0);
  rb_define_method(cls, "rb_security_error", sc_get_security_error, 0);
  rb_define_method(cls, "rb_not_imp_error", sc_get_not_imp_error, 0);
  rb_define_method(cls, "rb_no_mem_error", sc_get_no_mem_error, 0);
  rb_define_method(cls, "rb_script_error", sc_get_script_error, 0);
  rb_define_method(cls, "rb_syntax_error", sc_get_syntax_error, 0);
  rb_define_method(cls, "rb_load_error", sc_get_load_error, 0);
  rb_define_method(cls, "rb_system_call_error", sc_get_system_call_error, 0);
  rb_define_method(cls, "rb_errno", sc_get_errno, 0);
  rb_define_method(cls, "rb_io_error", sc_get_io_error, 0);
  rb_define_method(cls, "rb_local_jump_error", sc_get_local_jump_error, 0);
  rb_define_method(cls, "rb_eof_error", sc_get_eof_error, 0);
  rb_define_method(cls, "rb_thread_error", sc_get_thread_error, 0);
  rb_define_method(cls, "rb_zero_division_error", sc_get_zero_division_error, 0);
  rb_define_method(cls, "rb_float_domain_error", sc_get_float_domain_error, 0);
  rb_define_method(cls, "rb_sys_stack_error", sc_get_sys_stack_error, 0);
  rb_define_method(cls, "rb_regexp_error", sc_get_regexp_error, 0);
}

